lib/rust/ensogl/core/src/animation/frp/animation.rs

```diff
@@ -56,25 +61,30 @@ where mix::Repr<T>: inertia::Value
     pub fn new(network: &frp::Network) -> Self {
         frp::extend! { network
             value_src <- any_mut::<T>();
+            on_end_src <- any_mut();
         }
         let on_step = Box::new(f!((t) value_src.emit(mix::from_space::<T>(t))));
-        let simulator = AnimationSimulator::<T>::new(on_step, (), ());
-        // FIXME[WD]: The precision should become default and should be increased in all simulators
+        let on_end = Box::new(f!((_) on_end_src.emit(())));
+        let simulator = AnimationSimulator::<T>::new(on_step, (), on_end);
+        // FIXME[WD]: The precision should should be increased in all simulators
         //            that work with pixels. The reason is that by default the simulator should
         //            give nice results for animations in the range of 0 .. 1, while it should not
         //            make too many steps when animating bigger values (like pixels).
-        simulator.set_precision(0.001);
+        simulator.set_precision(DEFAULT_PRECISION);
         frp::extend! { network
             target    <- any_mut::<T>();
+            set_value <- any_mut::<T>();
             precision <- any_mut::<f32>();
             skip      <- any_mut::<()>();
             eval  target    ((t) simulator.set_target_value(mix::into_space(t.clone())));
+            eval  set_value ((t) simulator.set_value(mix::into_space(t.clone())));
```

Comment:
I don't like the idea for the ability to set the value of the physics-based animator. It can make the animations jumpy. It was not exposed on purpose. Can we have a fast call why this is needed and if there is a way not to expose it, please?

Reply:
Oh well, actually it turned out that I'm not using it anywhere :) It was used in some previous versions of the code